### PR TITLE
Parameterize with MM minimizers for initial minimization and dihedral optimization

### PR DESCRIPTION
**Pull request by stefdoerr**

### CODE REVIEW

File: htmd/parameterization/util.py [hunk 1]
```diff
@@ -172,28 +172,36 @@ def getFixedChargeAtomIndices(mol, fix_charge):
     return fixed_atom_indices
 
 
-def minimize(mol, qm, outdir):
+def minimize(mol, qm, outdir, min_type='qm', mm_minimizer=None):
```
Comment (raimis) [inline htmd/parameterization/util.py]: This looks ugly! We really need to refactor calculator and minimizer classes.

File: htmd/parameterization/util.py [hunk 2]
```diff
@@ -172,28 +172,36 @@ def getFixedChargeAtomIndices(mol, fix_charge):
...
+        qm.directory = mindir
+        results = qm.run()
+        if results[0].errored:
+            raise RuntimeError('\nQM minimization failed! Check logs at %s\n' % mindir)
+
+        mol = mol.copy()
+        # Replace coordinates with the minimized set
+        mol.coords = np.atleast_3d(np.array(results[0].coords, dtype=np.float32))
+    elif min_type == 'mm':
+        mol.coords[:, :, 0] = mm_minimizer.minimize(mol.coords)
```
[the remark this answers is not in the document]
Comment (stefdoerr) [reply htmd/parameterization/util.py]: fixed

File: htmd/parameterization/util.py [hunk 3]
```diff
@@ -230,7 +245,7 @@ def fitDihedrals(mol, qm, method, prm, all_dihedrals, dihedrals, outdir, geomopt
     for molecule, dihedral, directory in zip(molecules, dihedrals, directories):
         qm.molecule = molecule
         qm.esp_points = None
-        qm.optimize = geomopt
+        qm.optimize = True if dihed_opt_type == 'qm' else False
```
[the remark this answers is not in the document]
Comment (stefdoerr) [reply htmd/parameterization/util.py]: fixed

File: htmd/parameterization/util.py [hunk 4]
```diff
@@ -241,7 +256,7 @@ def fitDihedrals(mol, qm, method, prm, all_dihedrals, dihedrals, outdir, geomopt
...
```
[the remark this answers is not in the document]
Comment (stefdoerr) [reply htmd/parameterization/util.py]: fixed

File: htmd/parameterization/util.py (hunk 1 above)
Comment (stefdoerr) [reply htmd/parameterization/util.py]: agreed

**Comment by raimis** [comment]
Where is the `htmd-data` PR?

**Comment by j3mdamas** [comment]
I'll solve the conflicts and click buttons

**Comment by stefdoerr** [comment]
Otherwise we are  just getting stuck in a loop where we click the refresh buttons until one passes, we merge, it causes conflicts, we test again etc etc and it never gets merged.

**Comment by j3mdamas** [comment]
Wait, raimis is on it. Let's see.

**Comment by raimis** [comment]
Patience, my friend, there is nothing worse that a premature merge! :)

**Comment by raimis** [comment]
It will go after #548

**Comment by stefdoerr** [comment]
why?

**Pull request merged**